index.js

```diff
@@ -76,9 +76,15 @@ class ServerlessPlugin {
         }
 
         this.hooks = {
-            'before:package:setupProviderConfiguration': this.addCustomIamRoleForVendorArchiveDownload.bind(this),
+            'initialize': this.addCustomIamRoleForVendorArchiveDownload.bind(this),
```

Comment:
Isn't there a "package" event common to both ?

As far as I can tell, `initialize` is common to all commands so this would be executed for all commands (even unrelated ones like logs, invoke, etc).

Reply:
Very good point indeed.

That being said, this only adds an IAM role, so it should be fairly fast. And it ensures all commands have the same IAM role/config, so it might actually be a win? WDYT?

Reply:
Indeed @t-richard 🙂 

I tried to log every fired events in the issue: https://github.com/brefphp/bref/issues/1186#issuecomment-1212816753 (I can't be sure I got everything)

By the way, I never worked on a Serverless plugin, it's new territory for me 😅 

But I agree on the fact that only the necessary commands should setup the IAM permissions, even if it's a quick and small step.